GsaGH/Components/2_Geometry/Create2dElement.cs

```diff
@@ -46,7 +50,30 @@ protected override void SolveInstance(IGH_DataAccess da) {
         isLoadPanel = apiProperty2d.Type == Property2D_Type.LOAD;
       }
 
-      var elem = new GsaElement2d(ghmesh.Value, isLoadPanel);
+      GsaElement2d elem = null;
+      switch (geometry_parameter) {
+        case GH_Mesh mesh:
+          elem = new GsaElement2d(mesh.Value, isLoadPanel);
+          break;
+        case GH_Curve curve:
+          if (!isLoadPanel) {
+            throw new ArgumentException("Specify mesh geometry as the input parameter to create a 2D element.");
+          }
+          if (curve.Value.TryGetPolyline(out Rhino.Geometry.Polyline polyline)) {
+            if (polyline.ToArray().Length < 3) {
+              throw new ArgumentException("A minimum of three points are required to create a 2D element.");
+            }
+            elem = new GsaElement2d(polyline);
+          }
+          else {
+            throw new ArgumentException("Unable to retrieve polylines from the given curve");
+          }
+          break;
+        default: {
+            throw new ArgumentException("Input geometry is not supported to create a 2D element.");
+          }
+      }
+
```

Comment:
extract all of that to a GeometryValidation method that outputs a GsaElement2d.

The Exceptions, need to be set as a new class extending the closest Exception Type.

For instance

```csharp
public class InvalidGeometryForProperty : Exception
{
    public InvalidGeometryForProperty(GH_Curve curve): Exception("..."){}
    // You can under other constructors here for the other exceptions
   // Or create a new class, if they don't really fit
  
}
```